File: packages/stronghold_core/lib/models/responses/membership_package_response.dart
/// Matches backend MembershipPackageResponse exactly
class MembershipPackageResponse {
  final int id;
  final String? packageName;
  final double packagePrice;
  final String? description;

  const MembershipPackageResponse({
    required this.id,
    this.packageName,
    required this.packagePrice,
    this.description,
  });

  factory MembershipPackageResponse.fromJson(Map<String, dynamic> json) {
    return MembershipPackageResponse(
      id: (json['id'] ?? 0) as int,
      packageName: json['packageName'] as String?,
      packagePrice: ((json['packagePrice'] ?? 0) as num).toDouble(),
      description: json['description'] as String?,
    );
  }
}

File: Stronghold/Stronghold.Application/DTOs/Response/MembershipPackageResponse.cs
namespace Stronghold.Application.DTOs.Response
{
    public class MembershipPackageResponse
    {
        public int Id { get; set; }
        public string PackageName { get; set; } = string.Empty;
        public decimal PackagePrice { get; set; }
        public string Description { get; set; } = string.Empty;
    }
}
